### ISSUE TITLE
[bug] Build failing in Creating symlinks phase. No such file or directory *.dylib

### ISSUE THREAD

**Issue by CodingMeSwiftly** [leetal/ios-cmake#73]
**Describe the bug**
Build is failing during the "Creating symlinks" phase.

**To Reproduce**
1. Try to build LabStreamingLayer library https://github.com/sccn/liblsl
2. Add **BUNDLE DESTINATION ${LSLPREFIX}/bin** in liblsl/CMakeLists.txt#211
3. Execute **cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=../../ios.toolchain.cmake -DPLATFORM=OS64COMBINED** -> No issues.
4. Execute **cmake --build . --config Release --target install**
5. Build will run properly until link phase.
6. ** BUILD FAILED ** build/Release-iphoneos/liblsl64.dylib no such file or directory.

**Expected behavior**
There where no build issues prior to the linking phase, so I would expect .dylib to be there in the Release-iphoneos folder. But the file is missing, so linking fails with the above error.

**Environment**
* What version of CMake did you run?
cmake version 3.16.1
* What version of Xcode did you run?
Xcode Version 11.3 (11C29)
* Your faulty CMakeLists.txt here:
https://github.com/sccn/liblsl/blob/master/CMakeLists.txt

**Comment by leetal** [comment]
What happens if you add INSTALL_PREFIX with a path of your own choice to the cmake command, when generating the build targets?

This is probably an issue with the cmake configuration of liblsl rather than the toolchain itself.

**Comment by CodingMeSwiftly** [comment]
Thank you for your answer. I tried adding INSTALL_PREFIX as an option to the cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=../../ios.toolchain.cmake -DPLATFORM=OS64COMBINED command to no avail.
Unfortunately, I am very inexperienced with CMAKE and trying to solve the issue at hand is like searching for the needle in the haystack for me.
I don't want to ask too much, but I would appreciate it highly if you could try to build the liblsl library yourself. I guess you would be much faster in resolving the issues than I could ever be. Do you think you can take a shot at that in the next few days?

**Comment by leetal** [comment]
I'll look into it and see if I can spot the issue. 
A word of warning though. The combined build option is known to be buggy in larger projects (handled by cmake, not the toolchain). Sometimes it looks in the wrong directory for binaries, if the projects are split into multiple levels of cmake files.

**Comment by leetal** [comment]
So, i actually got it working by using the latest version of both the toolchain and the latest version (master) of liblsl. Not sure if that was needed but it is always best to use the latest  versions.. I used the following cmake command to skip the signing part: 
> cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=../ios.toolchain.cmake -DPLATFORM=OS64COMBINED -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED="NO" -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY=""

I also added the `BUNDLE DESTINATION ${LSLPREFIX}bin` to the install target in the CMakeLists.txt.

**Comment by CodingMeSwiftly** [comment]
First of all I really appreciate you spending time and looking into this!
I used the latest version of both tools and can now successfully build the liblsl library with the toolchain.
I found that in my previous attempts, in the CMAKELists.txt, I had added `BUNDLE DESTINATION ${LSLPREFIX}/bin` instead of `BUNDLE DESTINATION ${LSLPREFIX}bin` (mind the **/**). I guess that silly mistake caused the issue with the build script not finding some directories.
Well, nevertheless the issue is resolved.
Thank you again for your help.
